add naver map, LocationUpdateService

jooyul7013 opened the pull request:
add naver map, LocationUpdateService

running fragment 에 네이버 지도 연동
running fragment 생성 시 LocationUpdateService 바인딩

wooyoung-tom commented on feature/running/src/main/java/com/openrun/wantrunning/feature/running/service/LocationUpdateService.kt:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    private fun initData() {
+        locationRequest = LocationRequest.create()
+        locationRequest!!.interval = UPDATE_INTERVAL_IN_MILLISECONDS
+        locationRequest!!.priority = LocationRequest.PRIORITY_HIGH_ACCURACY
```
non-null assert 키워드 (`!!`) 를 사용하는것 보다 LocationRequest.create() 시에 `apply` scope function 을 사용하여 같이 초기화 해주는 것이 더 좋아보입니다! 🤔

jooyul7013 replied:
말씀주신 대로 수정하는 게 더 좋아보입니다! 수정 해서 반영했습니다 감사합니다 :)

wooyoung-tom commented on feature/running/src/main/java/com/openrun/wantrunning/feature/running/RunningViewModel.kt:
```diff
@@ -0,0 +1,42 @@
+@HiltViewModel
+class RunningViewModel @Inject constructor() : ViewModel() {
+    val isRunning = MutableLiveData(false)
+    lateinit var mNaverMap: NaverMap
```
ViewModel 외부에서도 `mNaverMap` 객체를 변경할 수 있어서 backing field 를 사용하거나 private setter 를 사용하는 것이 더 좋지 않을까 생각합니다.

jooyul7013 replied:
lateinit 이라 private getter setter 가 생성이안되어, setNaverMap() 이라는 함수형태로 만들어서 반영했습니다. 혹시 더 좋은 방법 있으시면 말씀부탁드립니다~

wooyoung-tom commented on feature/running/src/main/java/com/openrun/wantrunning/feature/running/RunningViewModel.kt:
```diff
@@ -0,0 +1,42 @@
+    val currentLocationOverlayCallback: LocationCallback = object : LocationCallback() {
+        override fun onLocationResult(locationResult: LocationResult) {
+            super.onLocationResult(locationResult)
+            val currentLocation: Location? = locationResult.lastLocation
+            val currentLatLng = LatLng(currentLocation?.latitude!!, currentLocation.longitude)
```
NPE 발생 가능성이 있어보입니다.
(저희 서비스에서도 드물지만 발생하고 있습니다 ㅠ)

jooyul7013 replied:
수정해서 반영했습니다. 감사합니다!

wooyoung-tom commented on feature/running/src/main/java/com/openrun/wantrunning/feature/running/RunningViewModel.kt:
```diff
@@ -0,0 +1,42 @@
+            val currentLocation: Location? = locationResult.lastLocation
+            val currentLatLng = LatLng(currentLocation?.latitude!!, currentLocation.longitude)
+
+            val cameraUpdate = CameraUpdate.scrollTo(currentLatLng)
+            mNaverMap.moveCamera(cameraUpdate)
+            val locationOverlay = mNaverMap.locationOverlay
+            if(!locationOverlay.isVisible) {
+                locationOverlay.isVisible = true
+                locationOverlay.position = currentLatLng
+            }
```
UI 를 업데이트 하는 코드라서 ViewModel 에 있는 것 보다는 Fragment 내에 있는 것이 적합하지 않을까 생각됩니다! 🤔

jooyul7013 replied:
UI 업데이트 하는 코드이긴 한데, NaverMap 객체를 통해 업데이트 하는 부분이라 어쩔수 없지 않을까 싶습니다. 해당 작업을 Fragment 내에서 수행하려면 NaverMap 객체를 Fragment 에서 관리하도록 해야할 것 같습니다. 회의때 얘기해보시죠^^

wooyoung-tom reviewed (approved):
고생하셨습니다! 👍 conflict resolve 는 다음 회의때 같이 하는걸로 하시죠!

Commit:
add naver map, LocationUpdateService, apply pr feedback

add naver map, LocationUpdateService
change Map Fragment to View

apply pr feedback
1. private setter
2. remove non-null assert and add null check
